ksql-parser/src/main/java/io/confluent/ksql/parser/ExpressionFormatter.java

```diff
@@ -235,7 +234,10 @@ protected String visitSymbolReference(SymbolReference node, Boolean context) {
     @Override
     protected String visitDereferenceExpression(DereferenceExpression node, Boolean unmangleNames) {
       String baseString = process(node.getBase(), unmangleNames);
-      return baseString + "." + formatIdentifier(node.getFieldName());
+      if (node.getBase() instanceof QualifiedNameReference) {
+        return baseString + "." + formatIdentifier(node.getFieldName());
+      }
+      return baseString + "->" + formatIdentifier(node.getFieldName());
```

Comment:
Should use constant.

Reply:
Already made them constant as @rodesai suggested too.